src/webpack-angular-types-plugin/type-extraction/type-extraction.ts

```diff
@@ -95,33 +95,53 @@ export function mergeProperties(
         const toMerge = properties[i];
         for (const inputToMerge of toMerge.inputs) {
-            // can happen if a newly defined input was defined as another property type
-            // e.g. base class defines @Output property, child class overrides it as in @Input() property
-            //      this should never happen in valid/useful angular code
-            removeFromMapIfExists(outputs, inputToMerge.name);
-            removeFromMapIfExists(
-                propertiesWithoutDecorators,
+            /*
+             *  can happen if a newly defined input was defined as another property type
+             * e.g. base class defines @Output property, child class overrides it as in @Input() property
+             *      this should never happen in valid/useful angular code
+             */
+            removeFromMapsIfExists(
+                [outputs, propertiesWithoutDecorators],
                 inputToMerge.name
             );
+            const existingInput = inputs.get(inputToMerge.name);
+            // there was already a getter/setter defined which holds a description/defaultValue
+            // therefore do not replace it for the docs here
+            if (
+                existingInput &&
+                existingInput.modifier &&
+                (existingInput.description || existingInput.defaultValue)
+            ) {
+                continue;
+            }
             inputs.set(inputToMerge.name, inputToMerge);
         }
         for (const outputToMerge of toMerge.outputs) {
-            removeFromMapIfExists(inputs, outputToMerge.name);
-            removeFromMapIfExists(
-                propertiesWithoutDecorators,
+            removeFromMapsIfExists(
+                [inputs, propertiesWithoutDecorators],
                 outputToMerge.name
             );
             outputs.set(outputToMerge.name, outputToMerge);
         }
         for (const propertyWithoutDecoratorsToMerge of toMerge.propertiesWithoutDecorators) {
-            removeFromMapIfExists(
-                inputs,
-                propertyWithoutDecoratorsToMerge.name
-            );
-            removeFromMapIfExists(
-                outputs,
+            removeFromMapsIfExists(
+                [inputs, outputs],
                 propertyWithoutDecoratorsToMerge.name
             );
+            const existingPropertyWithoutDecorators =
+                propertiesWithoutDecorators.get(
+                    propertyWithoutDecoratorsToMerge.name
+                );
+            // there was already a getter/setter defined which holds a description/defaultValue
+            // therefore do not replace it for the docs here
+            if (
+                existingPropertyWithoutDecorators &&
+                existingPropertyWithoutDecorators.modifier &&
+                (existingPropertyWithoutDecorators.description ||
+                    existingPropertyWithoutDecorators.defaultValue)
```

Comment:
You have this check multiple times. Maybe it's a good idea to extract the code in a separate function with a declarative name and get rid of most (or all) of the comments above?

Also, is it possible to spread something that is possibly undefined?

```
const { modifier, description, defaultValue } = existingPropertyWithoutDecorators;
if(modifier && (description || defaultValue))
```

this could help to make it more redable / shorter

Reply:
Hm you are right, looks kind of messy. I refactored it into a separate method, with a more detailed comment.